.cbmc-batch/jobs/aws_cryptosdk_cmm_generate_enc_materials/aws_cryptosdk_cmm_generate_enc_materials_harness.c

```diff
@@ -0,0 +1,135 @@
+    ensure_cryptosdk_edk_list_has_allocated_list(&materials->encrypted_data_keys);
+    __CPROVER_assume(aws_cryptosdk_edk_list_is_valid(&materials->encrypted_data_keys));
+
+    // Stub until https://github.com/diffblue/cbmc/issues/5344 is fixed
+    // edk_list Precondition: The list has valid list elements
+    /*
+      __CPROVER_assume(aws_cryptosdk_edk_list_elements_are_bounded(&materials->encrypted_data_keys, MAX_STRING_LEN));
+      ensure_cryptosdk_edk_list_has_allocated_list_elements(&materials->encrypted_data_keys);
+      __CPROVER_assume(aws_cryptosdk_edk_list_elements_are_valid(&materials->encrypted_data_keys));
+    */
+
+    // Set up the keyring trace
+    __CPROVER_assume(aws_array_list_is_bounded(
+        &materials->keyring_trace, MAX_NUM_ITEMS, sizeof(struct aws_cryptosdk_keyring_trace_record)));
+    __CPROVER_assume(materials->keyring_trace.item_size == sizeof(struct aws_cryptosdk_keyring_trace_record));
+    ensure_array_list_has_allocated_data_member(&materials->keyring_trace);
+    __CPROVER_assume(aws_array_list_is_valid(&materials->keyring_trace));
+    ensure_trace_has_allocated_records(&materials->keyring_trace, MAX_STRING_LEN);
+    __CPROVER_assume(aws_cryptosdk_keyring_trace_is_valid(&materials->keyring_trace));
+
+    *output = materials;
+    return AWS_OP_SUCCESS;
+}
+
+void aws_cryptosdk_cmm_generate_enc_materials_harness() {
+    const struct aws_cryptosdk_cmm_vt vtable = { .vt_size = sizeof(struct aws_cryptosdk_cmm_vt),
+                                                 .name    = ensure_c_str_is_allocated(SIZE_MAX),
+                                                 .destroy = nondet_voidp(),
+                                                 .generate_enc_materials =
+                                                     nondet_bool() ? generate_enc_materials : NULL,
+                                                 .decrypt_materials = nondet_voidp() };
+    __CPROVER_assume(aws_cryptosdk_cmm_vtable_is_valid(&vtable));
+
+    struct aws_cryptosdk_cmm *cmm = can_fail_malloc(sizeof(*cmm));
+    __CPROVER_assume(cmm);
+    cmm->vtable = &vtable;
+    __CPROVER_assume(aws_cryptosdk_cmm_base_is_valid(cmm));
+
+    struct aws_cryptosdk_enc_request *request = can_fail_malloc(sizeof(*request));
+    __CPROVER_assume(request);
+    request->alloc   = can_fail_allocator();
+    request->enc_ctx = can_fail_malloc(sizeof(*request->enc_ctx));
+    __CPROVER_assume(request->enc_ctx);
+    ensure_allocated_hash_table(request->enc_ctx, MAX_NUM_ITEMS);
+    __CPROVER_assume(aws_cryptosdk_enc_request_is_valid(request));
+
+    struct aws_cryptosdk_enc_materials **output = can_fail_malloc(sizeof(*output));
+    __CPROVER_assume(output);
+
+    // Run the function under test.
+    if (aws_cryptosdk_cmm_generate_enc_materials(cmm, output, request) == AWS_OP_SUCCESS) {
+        assert(aws_cryptosdk_enc_materials_is_valid(*output));
+    } else {
+        assert(*output == NULL);
+    }
```

Comment:
We should also check that `cmm` and `request` are valid.
```c
assert(aws_cryptosdk_cmm_base_is_valid(cmm));
assert(aws_cryptosdk_enc_request_is_valid(request));
```

Reply:
done